ground/src/main/java/com/google/android/ground/persistence/local/room/entity/LocationOfInterestEntity.java

```diff
@@ -130,9 +132,10 @@ public static LocationOfInterestEntity fromLocationOfInterest(
             .setCreated(AuditInfoEntity.fromObject(locationOfInterest.getCreated()))
             .setLastModified(AuditInfoEntity.fromObject(locationOfInterest.getLastModified()));
     if (locationOfInterest.getType() == LocationOfInterestType.POINT) {
-      entity.setLocation(Coordinates.fromPoint(locationOfInterest.getCoordinatesAsPoint()));
+      entity.setLocation(
```

Comment:
Do we still need `LocationOfInterestType`? Perhaps we can remove it in favor of `getGeometry()` and instanceof in the future?

Reply:
Once we have only kotlin callers, we can remove it and exclusively use `when` statements. I kept it around for the moment for java callers since it's a bit shorter than the long if/else instanceof statements.

Reply:
That makes sense! Would you mind filing a cleanup bug for that? 🙏

Reply:
will do!

Reply:
#1246